crates/storage/provider/src/traits/chain.rs

```diff
@@ -46,15 +46,15 @@ impl Stream for CanonStateNotificationStream {
     type Item = CanonStateNotification;
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        loop {
-            return match ready!(self.as_mut().project().st.poll_next(cx)) {
-                Some(Ok(notification)) => Poll::Ready(Some(notification)),
-                Some(Err(err)) => {
-                    debug!(%err, "canonical state notification stream lagging behind");
-                    continue
-                }
-                None => Poll::Ready(None),
+        return match ready!(self.as_mut().project().st.poll_next(cx)) {
```

Comment:
```suggestion
        return match ready!(self.as_mut().project().st.poll_next(cx))? {
```
I think you can even propagate the `Poll::Ready(None)` variant, and skip the match on `None`